cat/CaT-service.yaml

```diff
@@ -533,7 +536,9 @@ paths:
         content:
           application/json:
             schema:
-              $ref: '#/components/schemas/AgreementDetails'
+              allOf:
+                - $ref: '#/components/schemas/AgreementDetails'
+                - $ref: '#/components/schemas/EventRequest'
```

Comment:
The Open API generator seems to ignore these hybrid inline schema "types" - I'm just seeing if there's a flag we need to set to get it to produce the necessary classes - if not we may request they are moved into dedicated 'response' objects (otherwise we'll have to create the necessary model classes manually which defeats the point in using the generator)

Reply:
@davidhughes1-ccs I can't currently see a way to force the generator to come up with a wrapper class for these inline response types, would it be possible to extract them into dedicated objects in the `/components/schemas` section? Cheers!

Reply:
Addressed this in the commit below as well as the restructuring noted. @tommyb82